src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java

```diff
@@ -241,6 +246,32 @@ public void initializeFromItineraryAndQueryParams() throws IllegalArgumentExcept
         }
     }
 
+    /**
+     * Updates the mode in the OTP query parameter based on the modes from the itinerary.
+     */
+    private void updateModeInQueryParams() throws URISyntaxException {
+        Set<String> actualModes = itinerary.legs.stream()
+            .map(leg -> leg.mode)
+            .collect(Collectors.toSet());
+
+        // Remove WALK if other access modes are present (i.e. {BICYCLE|CAR|MICROMOBILITY}[_RENT]).
+        boolean hasAccessModes = actualModes.stream().anyMatch(mode -> {
+            String mainMode = mode.split("_")[0];
+            List<String> accessModes = List.of("BICYCLE", "CAR", "MICROMOBILITY");
+            return accessModes.contains(mainMode);
+        });
+
+        if (hasAccessModes) {
+            actualModes.remove("WALK");
+        }
```

Comment:
We won't ever want to remove the WALK mode for OTP query params as it must be there for stuff like Park and Ride where a user will need to walk after transit to the destination.

Reply:
I'll definitely add WALK back for the park and ride mode.

For other modes, here is a request that the UI sends to OTP without the WALK mode, is this incorrect?

![image](https://user-images.githubusercontent.com/56846598/99299045-a2c3d000-2818-11eb-8578-d00e7459059b.png)

Reply:
@evansiroky here is one instance where your recommendation will not work. On [this query](http://localhost:9966/#/?ui_activeSearch=57idfg1aw&ui_activeItinerary=3&fromPlace=3901%20El%20Rey%20Road%2C%20Orlando%2C%20FL%2C%20USA%3A%3A28.55452058786115%2C-81.42688751220705&toPlace=3354%20Curry%20Ford%20Road%2C%20Orlando%2C%20FL%2C%20USA%3A%3A28.52405843250429%2C-81.34088516235353&date=2020-11-16&time=14%3A57&arriveBy=false&mode=WALK%2CBUS%2CRAIL&showIntermediateStops=true&maxWalkDistance=1207&optimize=QUICK&walkSpeed=1.34&ignoreRealtimeUpdates=true&companies=&bannedRoutes=&numItineraries=3&otherThanPreferredRoutesPenalty=900&preferredRoutes=), if you try to monitor the bike-to-bus trip, ADDING "WALK" to the queried modes results in the OTP request not having the bike-to-bus itinerary, resulting in the itinerary check telling the itinerary is not possible. Please advise.

Reply:
aha, I can see that OTP has some additional logic about this that makes it so that the CAR_RENT, CAR_HAIL must have WALK added, but BICYCLE_RENT and MICROMOBILITY_RENT don't need that.  https://github.com/ibi-group/OpenTripPlanner/blob/8ec56c10702413f19bcea32e40d1ede0a18ee3f0/src/main/java/org/opentripplanner/api/parameter/QualifiedMode.java#L43-L80.

Reply:
Thanks for checking. I made the changes in https://github.com/ibi-group/otp-middleware/pull/104/commits/8dc8da655d6fd712bd4e40134d99c5cd2f844d8e including code to determine the correct CAR mode.